core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.ts

```diff
@@ -235,6 +237,14 @@ export class TopNavigationBarComponent implements OnInit, OnDestroy {
     // variable in place of the 'this' keyword.
     let that = this;
 
+    let currentPath = this.urlService.getPathname();
+    let backpathUrl = this.PAGES_WITHOUT_BACK_STATE.find(
+      PAGES_WITHOUT_BACK_STATE => currentPath.includes(PAGES_WITHOUT_BACK_STATE)
+    );
+    if (backpathUrl) {
+      this.navbackButtonUrl = backpathUrl.substring(0, backpathUrl.length - 1);
```

Comment:
Hm, why does this value depend on whether there is a match in the pathname?

Can we just direct the user to the previous page they were on if they click on the back button e.g. using `this.location.back()` (see https://angular.dev/api/common/Location#back)